src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -204,14 +203,37 @@ public void Save(string name)
             using (_connector.StartUserAction())
             {
                 Log.Debug($"Creating savepoint {name}", _connector.Id);
-                _connector.ExecuteInternalCommand($"SAVEPOINT {name}");
+                await _connector.ExecuteInternalCommand($"SAVEPOINT {name}", async);
             }
         }
 
         /// <summary>
-        /// Rolls back a transaction from a pending savepoint state.
+        /// Creates a transaction save point.
+        /// </summary>
+        public void Save(string name) => Save(name, false).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Creates a transaction save point.
+        /// </summary>
+        [PublicAPI]
+        public Task SaveAsync(string name, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            using (NoSynchronizationContextScope.Enter())
+                return Save(name, true);
+        }
+
+        /// <summary>
+        /// Creates a transaction save point.
         /// </summary>
-        public void Rollback(string name)
+        [PublicAPI]
+        public Task SaveAsync(string name) => SaveAsync(name, CancellationToken.None);
```

Comment:
Why not combine these two with a default argument?

Reply:
This is how public API is designed. Take a look at other methods in 'System.Data' or in 'Npgsql'. An explanation of that you can found [here](https://youtu.be/mAg2UGrkJdA?t=3893).

Reply:
Ah, interesting. In that case, please disregard!

Reply:
@YohDeadfall it's true that this is how System.Data is currently, but that's pretty old. The corefx guidelines have changed and it's not preferred to have a single method with an optional cancellation token parameter. The same is being adopted in https://github.com/dotnet/corefx/issues/35012, for example (this was communicated orally in a .NET design review meeting).